ktlint-core/src/main/kotlin/com/pinterest/ktlint/core/KtLint.kt

```diff
@@ -254,6 +294,10 @@ object KtLint {
         }
     }
 
+    private fun filterDisabledRules(rootNode: ASTNode, fqRuleId: String): Boolean {
+        return rootNode.getUserData(EDITOR_CONFIG_USER_DATA_KEY)?.disabledRules?.contains(fqRuleId) == false
```

Comment:
I'm probably missing something, but why is it looking at the editor config key while executing the rules? I thought it resolved any params from the editor config into a merged userData upstream of this?

Reply:
It's kind of roundabout, but the values in the userData map gets parsed into the members of the `EditorConfig` class. Now that you mention it, I'm not actually sure if this is required, or if we could just access the userData map directly. I was mostly trying to follow existing patterns. I think maybe it's not actually necessary because no individual rules will actually need to access the disabled rules, which is what it seems like the EditorConfig class is used for. I'll try to take a look tonight and see if I can simplify this code.

Reply:
I've moved it into its own key now.